designer/server/src/main/scala/pl/touk/nussknacker/ui/statistics/UsageStatisticsReportsSettings.scala

```diff
@@ -73,6 +80,33 @@ object UsageStatisticsReportsSettings {
     }.mkString("https://stats.nussknacker.io/?", "&", "")
   }
 
+  private def fingerprint(fingerprintFile: File): String = {
+    this.synchronized {
+      Option(cachedFingerprint).getOrElse {
+        cachedFingerprint = persistedFingerprint(fingerprintFile) {
+          randomFingerprint
+        }
+        cachedFingerprint
+      }
+    }
+  }
+
+  private[statistics] def invalidateCache(): Unit = {
+    this.synchronized {
+      cachedFingerprint = null
+    }
+  }
```

Comment:
I could do that but then I'll have to add some new class e.g. `UsageStatisticsReportsSettingsProvider`, because now it is simple: `UsageStatisticsReportsSettings` class + factory method so "class" is already reserved. I'll do it

Reply:
I added a new class and removed this invalidation. After all, var also disappears - there is only one, lazy val. Take a look again please